File: openapi-client/scalaz/src/main/scala/org/openapitools/client/api/MailStatsVolume.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import MailStatsVolume._

case class MailStatsVolume (
  /* Message counts keyed by destination (envelope `to`) email address. */
  to: Option[Map[String, Integer]],
/* Message counts keyed by sender (envelope `from`) email address. */
  from: Option[Map[String, Integer]],
/* Message counts keyed by originating client IP address. */
  ip: Option[Map[String, Integer]])

object MailStatsVolume {
  import DateTimeCodecs._

  implicit val MailStatsVolumeCodecJson: CodecJson[MailStatsVolume] = CodecJson.derive[MailStatsVolume]
  implicit val MailStatsVolumeDecoder: EntityDecoder[MailStatsVolume] = jsonOf[MailStatsVolume]
  implicit val MailStatsVolumeEncoder: EntityEncoder[MailStatsVolume] = jsonEncoderOf[MailStatsVolume]
}

File: swagger-client/dart/lib/model/mail_stats_volume.dart
part of swagger.api;

class MailStatsVolume {
  /* Message counts keyed by destination (envelope `to`) email address. */
  Map<String, int> to = {};
/* Message counts keyed by sender (envelope `from`) email address. */
  Map<String, int> from = {};
/* Message counts keyed by originating client IP address. */
  Map<String, int> ip = {};

  MailStatsVolume();

  @override
  String toString() {
    return 'MailStatsVolume[to=$to, from=$from, ip=$ip, ]';
  }

  MailStatsVolume.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    to = json['to'];
    from = json['from'];
    ip = json['ip'];
  }

  Map<String, dynamic> toJson() {
    return {
      'to': to,
      'from': from,
      'ip': ip
     };
  }

  static List<MailStatsVolume> listFromJson(List<dynamic> json) {
    return json == null ? new List<MailStatsVolume>() : json.map((value) => new MailStatsVolume.fromJson(value)).toList();
  }

  static Map<String, MailStatsVolume> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, MailStatsVolume>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new MailStatsVolume.fromJson(value));
    }
    return map;
  }
}
